tests/api/controllers/supervisor_field/supervisor_field.c

```diff
@@ -384,6 +384,22 @@ int main(int argc, char **argv) {
                          "Consecutive wb_supervisor_field_set_mf_float: second set instruction failed.");
   wb_robot_step(TIME_STEP);
 
+  // supervisor field update
+  self = wb_supervisor_node_get_self();
+  supervisor_field = wb_supervisor_node_get_field(self, "supervisor");
+  ts_assert_boolean_equal(wb_supervisor_field_get_sf_bool(supervisor_field), "'supervisor' field should be true");
+  ts_assert_boolean_equal(wb_robot_get_supervisor(), "'wb_robot_get_supervisor' field should return true");
+  wb_supervisor_field_set_sf_bool(supervisor_field, false);
+  wb_robot_step(TIME_STEP);
+  wb_robot_step(TIME_STEP);
```

Comment:
Do we really need two time steps here?

Reply:
Yes, this is required because since the field is updated after the end of the step we need a second step to get the update in the controller, but this is bad, I have an idea to improve this.

Reply:
This is now fixed, the supervisor state is now up to date after one single step.